posix/unistd.c

```diff
@@ -0,0 +1,606 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <openenclave/corelibc/errno.h>
+#include <openenclave/corelibc/limits.h>
+#include <openenclave/corelibc/pthread.h>
+#include <openenclave/corelibc/stdlib.h>
+#include <openenclave/corelibc/string.h>
+#include <openenclave/corelibc/sys/stat.h>
+#include <openenclave/corelibc/sys/utsname.h>
+#include <openenclave/corelibc/unistd.h>
+#include <openenclave/internal/posix/device.h>
+#include <openenclave/internal/posix/fdtable.h>
+#include <openenclave/internal/posix/raise.h>
+#include <openenclave/internal/time.h>
+#include <openenclave/internal/trace.h>
+#include "mount.h"
+#include "posix_t.h"
+
+int oe_gethostname(char* name, size_t len)
+{
+    int ret = -1;
+    struct oe_utsname uts;
+
+    if ((ret = oe_uname(&uts)) != 0)
+        OE_RAISE_ERRNO(oe_errno);
+
+    oe_strlcpy(name, uts.nodename, len);
+    ret = 0;
+
+done:
+    return ret;
+}
+
+int oe_getdomainname(char* name, size_t len)
+{
+    int ret = -1;
+    struct oe_utsname uts;
+
+    if ((ret = oe_uname(&uts)) != 0)
+        OE_RAISE_ERRNO(oe_errno);
+
+#ifdef _GNU_SOURCE
+    oe_strlcpy(name, uts.domainname, len);
+#else
+    oe_strlcpy(name, uts.__domainname, len);
```

Comment:
Isn't the point of the split that domainname is GNU-specific extension and not POSIX? __domainname is not populated in posix.c if not `_GNU_SOURCE` so this is just copying garbage values.

Reply:
I removed the **_GNU_SOURCE** check and now we always expect a field named **domainname**. This works with GCC and Clang on the enclave side. The OCALL for this fills in domainame for that platform. This does not imply necessarily that the platform supports such a structure with a domainname field. On Windows this field is filled in through other means.